sympy/printing/pycode.py

```diff
@@ -608,6 +608,13 @@ def _print_MatrixBase(self, expr):
             func = self._module_format('numpy.array')
         return "%s(%s)" % (func, self._print(expr.tolist()))
 
+    def _print_Identity(self, expr):
+        shape = expr.shape
+        if all([dim.is_Integer for dim in shape]):
+            return "%s(%s)" % (self._module_format('numpy.eye'), self._print(expr.shape[0]))
+        else:
+            raise NotImplementedError("Symbolic matrix dimensions are not supported")
```

Comment:
I don't think that it is covered in test cases. Can you write one using `raises`?

Reply:
Well, the test failings indicate that it should be allowed to print symbolic dimensions.
https://travis-ci.org/sympy/sympy/jobs/549301300#L4745

Reply:
I'm not sure I understand the error. Shouldn't the enclosing `raises` have stopped it?

Reply:
```suggestion
            raise NotImplementedError("Symbolic matrix dimensions are not yet supported for identity matrices")
```